Riot/Modules/Room/DataSources/RoomDataSource.m

```diff
@@ -481,39 +481,53 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
                                         upperDecorationView:upperDecorationView];
                     }
                     
-                    MXKReceiptSendersContainer* avatarsContainer;
+                    MXKReceiptSendersContainer* avatarsContainer = nil;
                     
                     // Handle read receipts (if any)
                     if (self.showBubbleReceipts && cellData.readReceipts.count && !isCollapsableCellCollapsed)
                     {
                         // Get the events receipts by ignoring the current user receipt.
                         NSArray* receipts =  cellData.readReceipts[component.event.eventId];
                         NSMutableArray *roomMembers;
+                        NSMutableArray *nonRoomMembers;
                         NSMutableArray *placeholders;
                         
                         // Check whether some receipts are found
                         if (receipts.count)
                         {
                             // Retrieve the corresponding room members
                             roomMembers = [[NSMutableArray alloc] initWithCapacity:receipts.count];
+                            nonRoomMembers = [[NSMutableArray alloc] initWithCapacity:receipts.count];
                             placeholders = [[NSMutableArray alloc] initWithCapacity:receipts.count];
                             
+                            avatarsContainer = [[MXKReceiptSendersContainer alloc] initWithFrame:CGRectMake(bubbleCell.frame.size.width - PlainRoomCellLayoutConstants.readReceiptsViewWidth + PlainRoomCellLayoutConstants.readReceiptsViewRightMargin, bottomPositionY + PlainRoomCellLayoutConstants.readReceiptsViewTopMargin, PlainRoomCellLayoutConstants.readReceiptsViewWidth, PlainRoomCellLayoutConstants.readReceiptsViewHeight) andMediaManager:self.mxSession.mediaManager];
+                            
                             for (MXReceiptData* data in receipts)
                             {
                                 MXRoomMember * roomMember = [self.roomState.members memberWithUserId:data.userId];
                                 if (roomMember)
                                 {
                                     [roomMembers addObject:roomMember];
                                     [placeholders addObject:[AvatarGenerator generateAvatarForMatrixItem:roomMember.userId withDisplayName:roomMember.displayname]];
+                                } else {
+                                    MXUser *user = [[MXUser alloc] initWithUserId:data.userId];
+                                    [nonRoomMembers addObject:user];
+                                    [placeholders addObject:[AvatarGenerator generateAvatarForMatrixItem:roomMember.userId withDisplayName:@" "]];
+                                    [user updateFromHomeserverOfMatrixSession:self.mxSession success:^{
```

Comment:
Doing network requests on a `cellForRowAtIndexPath` is definitely something you **don't** want to do.